### PR TITLE
ci: add fedora 34 and -fanalyzer build to ci checks

### PR DESCRIPTION
**Pull request by grondo**
This PR updates the flux-security automated CI build to build on Fedora Core 34. The address sanitizer build is updated to fedora 34, and to the normal build we add the `-fanalyzer` flag to `CFLAGS` which turns on gcc 11.2's build time static analysis.

This only found a couple issues (fixed here), but it will be nice to have this enabled for PRs going forward.

### CODE REVIEW

File: src/libca/sigcert.c [hunk 1]
```diff
@@ -368,15 +368,19 @@ int sigcert_store (const struct sigcert *cert, const char *name)
         goto error;
     if (sigcert_fwrite_public (cert, fp) < 0)
         goto error;
-    if (fclose (fp) < 0)
```
Comment (garlick) [inline src/libca/sigcert.c:371]: Good catch, `-fanalyer`!

There's a small typo in the commit message: "failse"

**Review by garlick — APPROVED** [review]
Nice, just once small commit message typo noticed.

File: src/libca/sigcert.c (hunk 1 above)
Comment (grondo) [reply src/libca/sigcert.c:371]: Thanks! Fixed and pushed.

**Comment by grondo** [comment]
Setting MWP.

**Comment by grondo** [comment]
I forgot to update the `.mergify.yml`. This will now need to be manually merged I think.

**Pull request merged**